ifrs17/Import/1ImportScope-Identities.ipynb

```diff
@@ -226,27 +226,28 @@
         {
             "cell_type": "code",
             "source": [
-                "public interface ValidAmountType : IScope<string, ImportStorage>",
+                "public interface ValidAmountType : IScope<(string DataNode, string EstimateType), ImportStorage>",
                 "\n{",
                 "\n    static ApplicabilityBuilder ScopeApplicabilityBuilder(ApplicabilityBuilder builder) =>",
                 "\n                builder.ForScope<ValidAmountType>(s => s.WithApplicability<BeAmountTypesFromIfrsVariables>(x => x.GetStorage().ImportFormat != ImportFormats.Cashflow ||",
-                "\n                                                                                                                x.GetStorage().IsSecondaryScope(x.Identity)));",
+                "\n                                                                                                                x.GetStorage().IsSecondaryScope(x.Identity.DataNode)));",
                 "\n    ",
-                "\n    IEnumerable<string> BeAmountTypes => GetStorage().GetRawVariables(Identity)",
+                "\n    IEnumerable<string> BeAmountTypes => GetStorage().GetRawVariables(Identity.DataNode)",
                 "\n                                            .Where(rv => rv.AmountType != null)",
                 "\n                                            .Select(x => x.AmountType)",
-                "\n                                            .Concat(GetStorage().DataNodeDataBySystemName[Identity].IsReinsurance ? (AmountTypes.CDR).RepeatOnce() : Enumerable.Empty<string>())",
+                "\n                                            // .Concat(GetStorage().DataNodeDataBySystemName[Identity.DataNode].IsReinsurance ? (AmountTypes.CDR).RepeatOnce() : Enumerable.Empty<string>())",
+                "\n                                            .Concat(GetStorage().DataNodeDataBySystemName[Identity.DataNode].IsReinsurance ? (Identity.EstimateType == EstimateTypes.TM ? (\"CDRAtInception\").RepeatOnce() : (AmountTypes.CDR).RepeatOnce()) : Enumerable.Empty<string>())",
```

Comment:
I would expect this logic to be applied later (at CSM/LC/TM scopes), like a filter on the amount type when you are in the TM context.

Reply:
By changing the Identity of `ValidAmountType` and adding this ternary operator to `BeAmountTypes`:
```c#
Identity.EstimateType == EstimateTypes.TM ? ("CDRAtInception").RepeatOnce() : (AmountTypes.CDR).RepeatOnce()) 
``` 
the idea was to make the scope EstimateType sensitive in order to use the CDRAtInception related logic only in the TechnicalMargin scope. Do you think that this `.Concat(...)` should be applied x3 in each of the TM,CSM and LC scopes?